### PR TITLE
[v2] Refactor PProf into agent service

### PR DESCRIPTION
**Pull request by BinaryFissionGames**
## Description of Changes
* Move pprof configuration into the config file
* Move pprof related code into the `service` package

## **Please check that the PR fulfills these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Add a changelog entry (for non-trivial bug fixes / features)
- [ ] CI passes

### CODE REVIEW

File: service/service.go [hunk 1]
```diff
@@ -19,13 +19,20 @@ type AgentService struct {
...
 func (a *AgentService) Start(s service.Service) error {
 	a.agent.Info("Starting stanza agent")
 	if err := a.agent.Start(a.persister); err != nil {
-		a.agent.Errorw("Failed to start stanza agent", zap.Any("error", err))
+		a.agent.Errorw("Failed to start stanza agent", zap.Error(err))
+		a.cancel()
+		return nil
+	}
+
+	if err := a.pprofProfiler.Start(); err != nil {
```
Comment (cpheps) [inline service/service.go:34]: Might be worth a comment here that profiling will actually only start if enabled. I had to scan back through to double check that was the case.

File: cmd/stanza/root.go [hunk 2]
```diff
@@ -86,114 +55,19 @@ func runRoot(command *cobra.Command, _ []string, flags *RootFlags) {
 	}()
 
 	// Build agent service
-	service, ctx, err := service.NewBuilder().
+	service, _, err := service.NewBuilder().
```
Comment (cpheps) [inline cmd/stanza/root.go:58]: Do we still need to even return a context? I see it's being ignored here. I think I originally just returned it for pprof. I'd be in favor or removing it from the return if it's not used.

File: service/service.go (hunk 1 above)
Comment (BinaryFissionGames) [reply service/service.go:34]: Good call, added a comment here and above the start function itself.

File: cmd/stanza/root.go (hunk 2 above)
Comment (BinaryFissionGames) [reply cmd/stanza/root.go:58]: Removed, since it doesn't seem like we use the context anywhere.

**Pull request merged**